### PR TITLE
Block Suggester improvements

### PR DESCRIPTION
**Pull request by teenoh**
## 🌟 What is the purpose of this PR?

This PR adds some improvements to the block suggester majorly around when to show the block suggester dropdown and properly ranking the search results.

## 🔗 Related links

- [Bug: @ menu hidden if text below?](https://app.asana.com/0/1202482456346915/1202510248434786) _(internal)_
- [Bug: attempt to only show / menu when user intends](https://app.asana.com/0/1202482456346915/1202518487775878) _(internal)_
- [Rank blocks in / menu by search query relevance](https://app.asana.com/0/1202482456346915/1202500568584715) _(internal)_

## 🔍 What does this change?

- The [method](https://github.com/hashintel/hash/blob/main/packages/hash/frontend/src/blocks/page/createSuggester/createSuggester.tsx#L33) used to find strings that bring up the block suggester didn't initially account for hard breaks. This [fixes](https://github.com/hashintel/hash/pull/740/commits/b6e350f3f4a7d2cfa2a6d71a41fe32626ec9da2a) that by replacing instances of a hard break with a space.
- Searching for blocks didn't rank results based on relevance. This was because block descriptions were also searched through. Decided to only search through block names for now
- Adds a UI when block/mention search returns no result

## 📜 Does this require a change to the docs?

- No

## ❓ How to test this?

1.  Checkout the branch
1.  Try actions shown in the "Before" demo
1.  Confirm they give the same results like in the "After" demo

## 📹 Demo

Before

https://user-images.githubusercontent.com/16928688/177162305-c9135f28-b3e3-4b82-9e0b-b9e0c028ce8c.mp4

After

https://user-images.githubusercontent.com/16928688/177162337-4ac95e4b-64b5-4026-a471-227adfc3326e.mp4

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202518487775878
  - https://app.asana.com/0/0/1202510248434786
  - https://app.asana.com/0/0/1202500568584715

### CODE REVIEW

**Review by kachkaev — APPROVED** [review]
Thanks @teenoh! I can confirm the fixes mentioned in asana tasks!

When I was playing with `@` mentions, I noticed a couple of more edge cases:

- Two subsequent mentions lose spacing in between
- The third mention overwrites the second one
- Dropdown menus are not triggered in the middle of multi-line text blocks.


https://user-images.githubusercontent.com/608862/177181584-264ef237-bc14-4f80-a0da-b3a7a99360dc.mp4

Marking this PR as approved anyway if you want to track and address the above observations separately. Happy to take another look if you want to push to the existing branch instead!

File: packages/hash/frontend/src/blocks/page/createSuggester/Suggester.tsx [hunk 1]
```diff
@@ -68,9 +68,7 @@ export const Suggester = <T,>({
           borderRadius: "6px",
           boxShadow:
             "0px 20px 41px rgba(61, 78, 133, 0.07), 0px 16px 25px rgba(61, 78, 133, 0.0531481), 0px 12px 12px rgba(61, 78, 133, 0.0325), 0px 2px 3.13px rgba(61, 78, 133, 0.02)",
-          borderWidth: `1px`,
-          borderStyle: "solid",
-          borderColor: palette.gray[20],
+          border: `1px solid ${palette.gray[20]}`,
```
Comment (kachkaev) [inline packages/hash/frontend/src/blocks/page/createSuggester/Suggester.tsx:71]: Optional: revert? (this change was necessary mid-way this PR, but is a noop now)

**Comment by nathggns** [comment]
Looking at this now

**Review by nathggns — APPROVED** [review]

**Comment by teenoh** [comment]
Thanks @kachkaev! I will address in a follow up task

File: packages/hash/frontend/src/blocks/page/createSuggester/Suggester.tsx (hunk 1 above)
Comment (teenoh) [reply packages/hash/frontend/src/blocks/page/createSuggester/Suggester.tsx:71]: Thanks. I think we can leave it as it is

**Pull request merged**